ios-swiftui/realm-template-apps-swiftui/TaskViews.swift

```diff
@@ -0,0 +1,141 @@
+import SwiftUI
+import RealmSwift
+
+/// Show a detail view of a task. User can edit or mark the task complete.
+struct TaskDetail: View {
+    // This property wrapper observes the Task object and
+    // invalidates the view when the Task object changes.
+    @ObservedRealmObject var task: Task
+    
+    var body: some View {
+        Form {
+            Section(header: Text("Edit Task Summary")) {
+                // Accessing the observed task object lets us update the live object
+                // No need to explicitly update the object in a write transaction
+                TextField("Summary", text: $task.summary)
+            }
+            Section {
+                Toggle(isOn: $task.isComplete) {
+                    Text("Complete")
+                }
+            }
+        }
+        .navigationBarTitle("Update Task", displayMode: .inline)
+    }
+}
+
+struct TaskRow: View {
+    @ObservedRealmObject var task: Task
+    
+    var body: some View {
+        NavigationLink(destination: TaskDetail(task: task)) {
+            Text(task.summary)
+            Spacer()
+            if task.isComplete {
+                Image(systemName: "checkmark")
+                    .foregroundColor(.blue)
+                    .padding(.trailing, 10)
+            }
+        }
+    }
+}
+
+/// Use views to see a list of all Tasks, add or delete tasks, or logout.
+struct TasksView: View {
+    // ObservedResults is a mutable collection; here it's
+    // all of the Task objects in the realm.
+    // You can append or delete tasks directly from the collection.
+    @ObservedResults(Task.self) var tasks
+    @State var addTask = false
+    @State var isLoggingOut = false
+    @State var taskSummary = ""
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                if addTask {
+                    NewTask(addTask: $addTask)
+                }
+                else if isLoggingOut {
+                    LogoutView(isLoggedOut: false, isLoggingOut: $isLoggingOut)
+                }
+                else {
+                    TaskList()
+                }
+            }
+            .navigationBarItems(leading: HStack {
+                Button("Logout") {
+                    isLoggingOut = true
```

Comment:
Check out the way this is done in the quick start app. The logout button is a special view that handles logging out, so any view can add logout functionality to the navigation bar items without needing to maintain its own state about logging out. The logout button can also be passed down from above in the hierarchy, so this view doesn't even to know that it has a logout button!